python/python/lance/ray/sink.py

```diff
@@ -15,10 +27,93 @@
 if TYPE_CHECKING:
     import pandas as pd
 
-__all__ = ["LanceDatasink"]
+__all__ = ["LanceDatasink", "LanceFragmentWriter", "LanceCommitter"]
+
+
+def _pd_to_arrow(
+    df: Union[pa.Table, "pd.DataFrame", Dict], schema: Optional[pa.Schema]
+) -> pa.Table:
+    """Convert a pandas DataFrame to pyarrow Table."""
+    from ..dependencies import _PANDAS_AVAILABLE
+    from ..dependencies import pandas as pd
+
+    if isinstance(df, dict):
+        return pa.Table.from_pydict(df, schema=schema)
+    if _PANDAS_AVAILABLE and isinstance(df, pd.DataFrame):
+        return pa.Table.from_pandas(df, schema=schema)
+    return df
+
+
+def _write_fragment(
+    stream: Iterable[pa.Table], uri: str, schema: Optional[pa.Schema] = None
+) -> Tuple[FragmentMetadata, pa.Schema]:
+    from ..dependencies import _PANDAS_AVAILABLE
+    from ..dependencies import pandas as pd
+
+    if schema is None:
+        first = next(stream)
```

Comment:
nit: can we use `peekable` from `more_itertools`? https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.peekable

Reply:
Dont feel that we should introduce a new deps for this line. Lets revisit this if we need more?

Reply:
sounds good. Just saying `more-itertools` might already be in our transitive dep, its a fairly common pacakge